File: repository/src/main/java/com/rxdroid/repository/cache/UserCache.java
package com.rxdroid.repository.cache;

import com.rxdroid.repository.model.User;

import java.util.concurrent.TimeUnit;

import io.reactivex.annotations.NonNull;

public class UserCache extends TimeLimitedCache<User> {

    private User user;

    public UserCache(@NonNull final long timeOut, @NonNull final TimeUnit timeUnit) {
        super(timeOut, timeUnit);
    }

    @Override
    public boolean isSameObjectCached(final User user) {
        return getData().equals(user);
    }

    @Override
    public void setData(final User data) {
        super.setData(data);
        this.user = data;
    }

    @Override
    public User getData() {
        return user;
    }

    @Override
    public boolean hasValidCachedData() {
        return getData() != null && isCachedTimingValid();
    }
}

File: repository/src/main/java/com/rxdroid/repository/cache/UserCache.kt
package com.rxdroid.repository.cache

import com.rxdroid.repository.model.User
import java.util.concurrent.TimeUnit

class UserCache(timeOut: Long, timeUnit: TimeUnit) : TimeLimitedCache<User>(timeOut, timeUnit) {

    private var user: User? = null

    override fun getData() = user

    override fun hasValidCachedData() = isTimingValid()

    override fun isSameObjectCached(t: User?) = getData() == t

}
